src/compressed_tensors/transforms/base.py

```diff
@@ -62,33 +61,47 @@ def __init__(
 
         :param transform: transform (e.g. torch.Tensor, scalar) to be applied
         """
-        if self.learnable:
-            self.transform = torch.nn.Parameter(transform.to(dtype).to(device))
-        else:
-            self.transform = torch.nn.Buffer(transform.to(dtype).to(device))
+        self.transform = torch.nn.Parameter(transform, requires_grad=False)
+        self.transform_name = transform_name
+        self.permutation = (
```

Comment:
> For randomized hadamards, splits up the math so that the underlying hadamard can be cached and the randomness is introduced as a separate permutation matrix

Since permutations are specific to `RandomHadamard`s, shouldn't we be implementing this logic on the `RandomHadamard` class, not the general `Transforms` class?